incubator/hnc/pkg/forest/forest.go

```diff
@@ -197,6 +202,33 @@ func (ns *Namespace) clean() {
 	delete(ns.forest.namespaces, ns.name)
 }
 
+// Deleting returns true if the namespace is being deleted.
+func (ns *Namespace) Deleting() bool {
+	return ns.deleting
+}
+
+// SetDeleting marks this namespace as being deleted.
+func (ns *Namespace) SetDeleting() {
+	ns.deleting = true
+}
+
+// UpdateAllowCascadingDelete updates if this namespace allows cascading deletion.
+func (ns *Namespace) UpdateAllowCascadingDelete(acd bool) {
+	ns.allowCascadingDelete = acd
+}
+
+// AllowCascadingDelete returns if the namespace's or any of the owner ancestors'
+// allowCascadingDelete field is set to true.
+func (ns *Namespace) AllowCascadingDelete() bool {
```

Comment:
Nit: this looks like a command. Maybe consider `AllowsCascadingDelete` or `CascadingDeletionAllowed`